### ISSUE TITLE
feature request: increase number of allowed loops in gpioWaveChain

### ISSUE THREAD

**Issue by korroktheslavemaster** [joan2937/pigpio#382]
Right now there is a hard limit of 20 loops (as per documentation) in a wave chain
I am trying to create a ramp pulse input with ~100 steps of increasing frequencies (5000us to 10us), however the library only allows 20 steps at most. Is this a hardware limitation or can it be changed?
In the code I can see the limit is defined as 
```
#define WCB_COUNTERS (WCB_CNT_PER_PAGE * PI_WAVE_COUNT_PAGES)
```
where `WCB_CNT_PER_PAGE` is 2 and `PI_WAVE_COUNT_PAGES` is 10
Any help would be appreciated.

**Comment by guymcswain** [comment]
Where are you seeing this limitation?  The documentation, as I read it, shows a loop count of 
```
loop x + y*256 times
```
is possible.  Both x, y are 8-bit integers.

**Comment by korroktheslavemaster** [comment]
sorry i meant loop **counters**, not loop count
```
The code is currently dimensioned to support a chain with roughly 600 entries and 20 loop counters.
```
i.e. i can have only 20 loop blocks in the chain, having more throws `PI_CHAIN_COUNTER`

**Comment by guymcswain** [comment]
Ok, thanks for clarifying.  The limitation may be due to the allocated dma control blocks in memory which may have been arbitrary when @joan2937 implemented the code.  Even if it is, to change it could require a fairly involved effort.

**Comment by korroktheslavemaster** [comment]
Thank you. I'm up for contributing if @joan2937 thinks this is worthwhile, or I'd like to know if there's another way around this.

**Comment by korroktheslavemaster** [comment]
I couldn't find any documentation related to sending synchronous wave chains. Do you mean using `wave_send_using_mode(PIN, pigpio.WAVE_MODE_ONE_SHOT_SYNC)` to send individual waves?

**Comment by guymcswain** [comment]
Yes.  But now that I think about it, you would have to send each individual wave.  If they are 20ms long and your network isn't slow, you should be able to keep up using `gpiowavetxat` to know when the next wave is playing.
